tests/gadget-zero/main-stm32f429i-disco.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2015 Karl Palsson <karlp@tweak.net.au>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/rcc.h>
+
+#include <stdio.h>
+#include "usb-gadget0.h"
+
+#define ER_DEBUG
+#ifdef ER_DEBUG
+#define ER_DPRINTF(fmt, ...) \
+    do { printf(fmt, ## __VA_ARGS__); } while (0)
+#else
+#define ER_DPRINTF(fmt, ...) \
+    do { } while (0)
+#endif
+
+
+//#define USE_USB_INTERRUPT
+
+
+static usbd_device *usbd_dev;
+
+#ifdef USE_USB_INTERRUPT
+
+void otg_hs_isr(void)
+{
+	usbd_poll(usbd_dev);
+}
+
+#endif
+
+
+int main(void)
+{
+	rcc_clock_setup_hse_3v3(&rcc_hse_8mhz_3v3[RCC_CLOCK_3V3_168MHZ]);
+	rcc_periph_clock_enable(RCC_GPIOB);
+	rcc_periph_clock_enable(RCC_OTGHS);
+
+	gpio_mode_setup(GPIOB, GPIO_MODE_AF, GPIO_PUPD_NONE,
+			GPIO13 | GPIO14 | GPIO15);
+	gpio_set_af(GPIOB, GPIO_AF12, GPIO13 | GPIO14 | GPIO15);
+
+	/* LEDS on discovery board */
+	rcc_periph_clock_enable(RCC_GPIOD);
+	gpio_mode_setup(GPIOD, GPIO_MODE_OUTPUT,
+					GPIO_PUPD_NONE, GPIO12 | GPIO13 | GPIO14 | GPIO15);
+
+	usbd_dev = gadget0_init(&otghs_usb_driver, "stm32f429i-disco");
+
+	ER_DPRINTF("bootup complete\n");
+	
+#ifndef USE_USB_INTERRUPT
+	while (1) {
+		usbd_poll(usbd_dev);
+	}
+#else
+	nvic_enable_irq(NVIC_OTG_HS_IRQ);
+	
+	volatile int i = 1;
+	while (i);
```

Comment:
I think the standard "while (1) {} " construct is pretty invincible, and compiles always to an endless loop.